pytensor/tensor/slinalg.py

```diff
@@ -791,8 +797,8 @@ def make_node(self, A, B):
     def perform(self, node, inputs, output_storage):
         (A, B) = inputs
         X = output_storage[0]
-
-        X[0] = scipy.linalg.solve_continuous_lyapunov(A, B)
+        out_dtype = pytensor.scalar.upcast(A.dtype, B.dtype)
+        X[0] = scipy.linalg.solve_continuous_lyapunov(A, B).astype(out_dtype)
```

Comment:
`dtype` should be the one defined in `node.outputs[0].type.dtype`.

Same below

Reply:
I was hoping there was a better way to do that